Maple2.File.Parser/TableParser.cs

```diff
@@ -603,123 +603,113 @@ public class TableParser {
         Debug.Assert(data != null);
-
-        foreach (IndividualItemDrop drop in data.individualDropBox) {
-            yield return (drop.individualDropBoxID, drop);
+        foreach (IGrouping<byte, IndividualItemDrop> group in data.individualDropBox.GroupBy(dropbox => dropbox.dropGroup)) {
+            yield return (group.Key, group);
         }
     }
 
-    public IEnumerable<(int Id, IndividualItemDrop ItemDrop)> ParseIndividualItemDropQuestMob() {
+    public IEnumerable<(int Id, IEnumerable<IndividualItemDrop> ItemDrops)> ParseIndividualItemDropQuestMob() {
         string xml = Sanitizer.RemoveEmpty(xmlReader.GetString(xmlReader.GetEntry("table/individualitemdrop_quest_mob.xml")));
         xml = Sanitizer.SanitizeBool(xml);
         var reader = XmlReader.Create(new StringReader(xml));
         var data = individualItemDropSerializer.Deserialize(reader) as IndividualItemDropRoot;
         Debug.Assert(data != null);
-
-        foreach (IndividualItemDrop drop in data.individualDropBox) {
-            yield return (drop.individualDropBoxID, drop);
+        foreach (IGrouping<byte, IndividualItemDrop> group in data.individualDropBox.GroupBy(dropbox => dropbox.dropGroup)) {
+            yield return (group.Key, group);
         }
     }
 
-    public IEnumerable<(int Id, IndividualItemDrop ItemDrop)> ParseIndividualItemDropGacha() {
+    public IEnumerable<(int Id, IEnumerable<IndividualItemDrop> ItemDrops)> ParseIndividualItemDropGacha() {
         string xml = Sanitizer.RemoveEmpty(xmlReader.GetString(xmlReader.GetEntry("table/na/individualitemdrop_gacha.xml")));
         xml = Sanitizer.SanitizeBool(xml);
         var reader = XmlReader.Create(new StringReader(xml));
         var data = individualItemDropSerializer.Deserialize(reader) as IndividualItemDropRoot;
         Debug.Assert(data != null);
-
-        foreach (IndividualItemDrop drop in data.individualDropBox) {
-            yield return (drop.individualDropBoxID, drop);
+        foreach (IGrouping<byte, IndividualItemDrop> group in data.individualDropBox.GroupBy(dropbox => dropbox.dropGroup)) {
+            yield return (group.Key, group);
         }
     }
 
-    public IEnumerable<(int Id, IndividualItemDrop ItemDrop)> ParseIndividualItemGearBox() {
+    public IEnumerable<(int Id, IEnumerable<IndividualItemDrop> ItemDrops)> ParseIndividualItemGearBox() {
         string xml = Sanitizer.RemoveEmpty(xmlReader.GetString(xmlReader.GetEntry("table/na/individualitemdrop_gearbox.xml")));
         xml = Sanitizer.SanitizeBool(xml);
         var reader = XmlReader.Create(new StringReader(xml));
         var data = individualItemDropSerializer.Deserialize(reader) as IndividualItemDropRoot;
         Debug.Assert(data != null);
-
-        foreach (IndividualItemDrop drop in data.individualDropBox) {
-            yield return (drop.individualDropBoxID, drop);
+        foreach (IGrouping<byte, IndividualItemDrop> group in data.individualDropBox.GroupBy(dropbox => dropbox.dropGroup)) {
```

Comment:
This seems a bit wrong, you are returning the `group.Key` as `int Id` and missing `drop.individualDropBoxID` entirely. Wouldn't it be better to return a dictionary so you can preserve this grouping?

```cs
public IEnumerable<(int id, IDictionary<byte, IList<IndividualItemDrop>>)> Parse();
```

Reply:
oh I see. is there a simplified way to foreach loop this? havent tried doing something like that before

Reply:
Untested but maybe something like:

```cs
Dictionary<byte, IList<IndividualItemDrop>> groups = data.individualDropBox.GroupBy(dropbox => dropbox.dropGroup)
    .ToDictionary(group => group.Key, group => group.ToList());
yield return (drop.individualDropBoxID, groups)
```